src/metabase/search/postgres/ingestion.clj

```diff
@@ -69,26 +70,33 @@
     (let [as (keyword (u/->snake_case_en (name k)))]
       (if (true? v) as [v as]))))
 
-;; TODO memoize
-(defn- spec-index-query [search-model]
+(defn- spec-index-query*
+  [search-model]
   (let [spec (search.spec/spec search-model)]
     (u/remove-nils
-     {:select (search.spec/qualify-columns :this
-                                           (concat
-                                            (:search-terms spec)
-                                            (mapcat (fn [k] (attrs->select-items (get spec k)))
-                                                    [:attrs :render-terms])))
-      :from   [[(t2/table-name (:model spec)) :this]]
-      :where  (:where spec [:inline [:= 1 1]])
+     {:select    (search.spec/qualify-columns :this
+                                              (concat
+                                               (:search-terms spec)
+                                               (mapcat (fn [k] (attrs->select-items (get spec k)))
+                                                       [:attrs :render-terms])))
+      :from      [[(t2/table-name (:model spec)) :this]]
+      :where     (:where spec [:inline [:= 1 1]])
       :left-join (when (:joins spec)
                    (into []
                          cat
                          (for [[join-alias [join-model join-condition]] (:joins spec)]
                            [[(t2/table-name join-model) join-alias]
                             join-condition])))})))
 
-(defn- spec-index-reducible [search-model]
-  (->> (spec-index-query search-model)
+(def ^{:private true, :arglists '([search-model])} spec-index-query
```

Comment:
should we disable memoization during dev?

if not why this can't be a `def` or `defonce`?

Reply:
Hmm, not sure how we would memoize for dev. Using a def instead of defonce is close enough for me, because then if I make a change to this function I just re-evaluate the whole buffer.

Reply:
you can add ` ::memoize/args-fn` function that returns random number in dev mode.

but if a `def` or `defonce` is what you need then I'd vote for going that way.